core/trino-main/src/main/java/io/trino/sql/planner/plan/JoinNode.java

```diff
@@ -42,6 +42,12 @@
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * NOTE WELL: SQL MERGE requires that all rows that have the same rowId and
```

Comment:
```
SQL MERGE requires that all rows that have the same rowId and
 * redistribution columns be sent to the same node.
```
Does it require special comment? Essentially `rowId` is a join key.
Do you actually mean that SQL MERGE only works with repartition join?

BTW: It seems this comment belongs somewhere else (e.g. pherhaps `AddExchange`) rather than generic `JoinNode`

Reply:
We'll delete the comment